galasa-managers-parent/galasa-managers-core-parent/dev.galasa.textscan.manager/src/main/java/dev/galasa/textscan/ITextScanner.java

```diff
@@ -0,0 +1,422 @@
+     * @return This text scanner for fluent calls
+     * @throws FailTextFoundException - If the failurePattern was found
+     * @throws MissingTextException - If no occurrences of the searchPattern was found 
+     */
+    ITextScanner scan(InputStream inputStream, Pattern searchPattern, Pattern failPattern) throws FailTextFoundException, MissingTextException;
+    
+    /**
+     * Convenience method for scan(inputStream, searchPattern, null, 1)
+     * 
+     * @param inputStream - The inputStream  to be searched
+     * @param searchPattern - The regex to search for
+     * @return This text scanner for fluent calls
+     * @throws MissingTextException - If no occurrences of the searchPattern was found 
+     */
+    ITextScanner scan(InputStream inputStream, Pattern searchPattern) throws FailTextFoundException, MissingTextException;
+    
+    /**
+     * Convenience method for scan(inputStream, Pattern.Compile("\Q" + searchString + "\E"), Pattern.Compile("\Q" + failString + "\E"), count)
+     * 
+     * @param inputStream - The inputStream to be searched
+     * @param searchText - The text to search for
+     * @param failText - Failure text to search for, can be null meaning no fail search
+     * @param count - Atleast how many occurrences of the searchText must exist
+     * @return This text scanner for fluent calls
+     * @throws FailTextFoundException - If the failureText was found
+     * @throws MissingTextException - If no occurrences of the searchText was found 
+     * @throws IncorrectOccurancesException - If insufficient occurrences were found, if there are zero occurrences, then MissingTextException will be thrown
+     */
+    ITextScanner scan(InputStream inputStream, String searchString, String failString, int count) throws FailTextFoundException, MissingTextException, IncorrectOccurancesException;
+    
+    /**
+     * Convenience method for scan(inputStream, searchText, failText, 1)
+     * 
+     * @param inputStream - The inputStream to be searched
+     * @param searchText - The text to search for
+     * @param failText - Failure text to search for, can be null meaning no fail search
+     * @return This text scanner for fluent calls
+     * @throws FailTextFoundException - If the failureText was found
+     * @throws MissingTextException - If no occurrences of the searchText was found 
+     */
+    ITextScanner scan(InputStream inputStream, String searchString, String failString) throws FailTextFoundException, MissingTextException;
+
+    /**
+     * Convenience method for scan(inputStream, searchText, null, 1)
+     * 
+     * @param inputStream - The inputStream to be searched
+     * @param searchText - The text to search for
+     * @return This text scanner for fluent calls
+     * @throws MissingTextException - If no occurrences of the searchText was found 
+     */
+    ITextScanner scan(InputStream inputStream, String searchString) throws FailTextFoundException, MissingTextException;
+
```

Comment:
REALLY small thing - but can we kill the excess blank lines

Reply:
it was so show the grouping of methods.

Reply:
fair - might I suggest a multi-line comment to make it more obvious?

Reply:
true,  removed for the moment as the overloaded methods removed, temporarily